pkg/chain/ethereum/tbtc.go

```diff
@@ -374,3 +414,37 @@ func (tec *TBTCEthereumChain) getDepositContract(
 
 	return depositContract, nil
 }
+
+// DepositAddressForKeepAddress retrieves the deposit address for a particular keep address
+func (tec *TBTCEthereumChain) DepositAddressForKeepAddress(
+	keepAddress string,
+) (string, error) {
+	createdEvents, err := tec.PastCreatedEvents(0, keepAddress)
```

Comment:
An alternative way of obtaining a deposit address for a keep could be calling the keep's `getOwner()` function. When creating a keep the deposit should set itself as the owner of the keep.

Maybe this would be more efficient?

Reply:
That sounds *much* more efficient. How confident are we that that's always true?

Reply:
I'm 99,99999% sure. @Shadowfiend @pdyraga do you guys see any use case where this won't be true?

Reply:
Depends on Ethereum deployment. If it's a self-hosted Ethereum client, a single call for targeted information is more efficient than pulling events from logs. If it's Infura, (a selfish :wink:) operator does not care because both are just one request. If it is Alchemy, they charge for compute units so a single call will be better.

Reply:
It sounds like using `getOwner()` is never worse and sometimes better, which means we should use it!